portfolio/src/main/java/com/google/sps/servlets/ContactServlet.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/contact")
+public class ContactServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String name = getParameter(request, "name-input", "");
+    String email = getParameter(request, "email-input", "");
+    String phone = getParameter(request, "phone-input", "");
+    long timestamp = System.currentTimeMillis();
+    if(email.strip() == "" && phone.strip()== "") {
+      response.setContentType("text/html;");
+      response.getWriter().println("You have not entered valid info, please go back and re-fill the form");
+      return;
+    }
+    Entity contactEntity = new Entity("Contact");
+    contactEntity.setProperty("Name", name);
+    contactEntity.setProperty("E-Mail", email);
+    contactEntity.setProperty("Phone", phone);
+    contactEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(contactEntity);
+    boolean emailContact = Boolean.parseBoolean(getParameter(request, "email", "false"));
+    boolean phoneContact = Boolean.parseBoolean(getParameter(request, "phone", "false"));
```

Comment:
I'm curious if this is working as intended? My understanding is that `parseBoolean` returns true if the string argument is not null and is equal, ignoring case, to the string "true". For example, if the email is example@gmail.com, this boolean would be be false (but it seems like we'd want it to be true). Let me know if I'm misunderstanding the code!

Reply:
Yes, I was able to get it working as intended. My statement receives a response from a checkbox and the "getParameter" function ensures an appropriate boolean is returned.

Reply:
Cool, thanks for checking!